lib/Sema/MiscDiagnostics.cpp

```diff
@@ -59,8 +59,8 @@ using namespace constraints;
 /// Return true if this expression is an implicit promotion from T to T?.
 static Expr *isImplicitPromotionToOptional(Expr *E) {
   if (E->isImplicit())
-    if (auto IIOE = dyn_cast<InjectIntoOptionalExpr>(
-                                               E->getSemanticsProvidingExpr()))
+    if (auto IIOE =
+            dyn_cast<InjectIntoOptionalExpr>(E->getSemanticsProvidingExpr()))
```

Comment:
Formatting changes unrelated to the fix such as this one should be omitted or extracted into a separate PR.